pkg/connector/courses.go

```diff
@@ -193,6 +195,63 @@ func (o *courseBuilder) Grants(ctx context.Context, resource *v2.Resource, pToke
 	return rv, nextPageToken, nil, nil
 }
 
+func (o *courseBuilder) Grant(ctx context.Context, principal *v2.Resource, entitlement *v2.Entitlement) ([]*v2.Grant, annotations.Annotations, error) {
+	l := ctxzap.Extract(ctx)
+
+	if principal.Id.ResourceType != userResourceType.Id {
+		l.Warn(
+			"litmos-connector: only users can be granted course entitlement",
+			zap.String("principal_type", principal.Id.ResourceType),
+			zap.String("principal_id", principal.Id.Resource),
+		)
+		return nil, nil, fmt.Errorf("litmos-connector: only users can be granted course entitlement")
```

Comment:
Minor nitpick: It'd be nice if these errors used grpc statuses of INVALID_ARGUMENT. It won't change the behavior of baton-sdk or c1, but it's nice to classify errors if we can.